dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePluginFunctionalTest.kt

```diff
@@ -0,0 +1,323 @@
+        // The assembleRelease fails for configuration cache and not because of our
+        // code but because of these tasks: collect[Flavour1][Flavour2]ReleaseDependencies
+
+        // This test should fail because our `DdCheckSdkDepsTask` is not following the
+        // --configuration-cache best practices (we are using the project inside) and so far we
+        // could not easily solve that.
+
+        // Given
+        stubGradleBuildFromResourceFile("build_with_datadog_dep.gradle.kts")
+
+        // When
+        assertThatThrownBy {
+            GradleRunner.create()
+                .withProjectDir(testProjectDir)
+                .withArguments("--configuration-cache", "assemble")
+                .withPluginClasspath(getTestConfigurationClasspath())
+                .build()
+        }.isInstanceOf(UnexpectedBuildFailure::class.java)
+    }
+
+    @Test
+    fun `M success W assembleDebug { configuration cache, checkProjectDependencies disabled }`() {
+
+        // Given
+        stubGradleBuildFromResourceFile("build_with_check_deps_disabled.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments("--configuration-cache", "assembleDebug")
+            .withPluginClasspath(getTestConfigurationClasspath())
+            .build()
+
+        // Then
+        assertThat(result.task(":assembleDebug")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M success W assemble { Datadog SDK not in deps, checkProjectDepedencies set to warn }`() {
+        // Given
+        stubGradleBuildFromResourceFile("build_with_check_deps_set_to_warn.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments("assemble")
+            .withPluginClasspath(getTestConfigurationClasspath())
+            .build()
+
+        // Then
+        assertThat(result.task(":assemble")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M fail W assemble { Datadog SDK not in deps }`() {
+        // Given
+        stubGradleBuildFromResourceFile("build_without_datadog_dep.gradle.kts")
+        // When
+        assertThatThrownBy {
+            GradleRunner.create()
+                .withProjectDir(testProjectDir)
+                .withArguments("assemble")
+                .withPluginClasspath(getTestConfigurationClasspath())
+                .build()
+        }.isInstanceOf(UnexpectedBuildFailure::class.java)
+    }
+
+    // endregion
+
+    // region Upload
+
+    @ParameterizedTest
+    @MethodSource("uploadTasks")
+    fun `M success W upload`(taskName: String) {
+        // Given
+        stubGradleBuildFromResourceFile("build_with_datadog_dep.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments(taskName)
```

Comment:
without calling `assembleXXX` there won't be mapping file and task will be effectively no-op according to this line https://github.com/DataDog/dd-sdk-android-gradle-plugin/blob/ba2d61297317a004efab56665257dcdbaff9e0be/dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/DdMappingFileUploadTask.kt#L106.

Mapping Task is independent and doesn't call `assemble` by itself.

Also maybe we need to add real files/modules to the `src/main`? Not sure, but it it might be that without our files there won't be any mapping file.

Reply:
There is more than that actually, it needs to be a git versioned directory otherwise it will not work (I think you already did something like that for `GitRepositoryDetectorTest`). I will have a look into this and try to provide a dummy Application class in the `src/main`